cfme/tests/control/test_actions.py

```diff
@@ -49,11 +50,27 @@ def __getattr__(self, key):
         return partial(func, self._vm)
 
 
-pytest_generate_tests = testgen.generate(
-    testgen.infra_providers,
-    "small_template",
-    scope="module"
-)
+def pytest_generate_tests(metafunc):
+    # Filter out providers without provisioning data or hosts defined
+    argnames, argvalues, idlist = testgen.infra_providers(metafunc,
+        'small_template', scope="module")
+
+    new_idlist = []
+    new_argvalues = []
+    for i, argvalue_tuple in enumerate(argvalues):
+        args = dict(zip(argnames, argvalue_tuple))
+
+        if ((metafunc.function is test_action_create_snapshot_and_delete_last)
+            or
+            (metafunc.function is test_action_create_snapshots_and_delete_them)) \
+                and isinstance(args['provider_crud'], RHEVMProvider):
+            continue
```

Comment:
I wonder if we could make some kind of uncollect mark that would enable us to not write custom generate_tests hooks ...